src/Microsoft.Health.Dicom.Core/Features/Validation/TagPathValidator.cs

```diff
@@ -0,0 +1,22 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Text.RegularExpressions;
+using Microsoft.Health.Dicom.Core.Exceptions;
+
+namespace Microsoft.Health.Dicom.Core.Features.Validation
+{
+    public static class TagPathValidator
```

Comment:
>public [](start = 4, length = 6)

Does this class need to be `public`? This is more of a general question/comment across the PR. We should be judicious about what we decide to be `public`. `public` things become part of the assembly's contract, and can't be changed as easily. As a rule of thumb, only make what is absolutely necessary `public`/`protected`. Everything else can be `internal`/`private`.

Reply:
Hmm, I was following the pattern used in UidValidator. I think it's public also because of accessibility in test code?

---
In reply to: [564837026](https://github.com/microsoft/dicom-server/pull/501#discussion_r564837026) [](ancestors = 564837026)

Reply:
You can explicitly share your `internal` components with other assemblies, typically test assemblies, using the below attribute in your *AssemblyInfo.cs*:

```csharp
[assembly: InternalsVisibleTo("Microsoft.Health.Dicom.UnitTests")]
```

---
In reply to: [566994328](https://github.com/microsoft/dicom-server/pull/501#discussion_r566994328) [](ancestors = 566994328,564837026)